pkg/service/topology/topology.go

```diff
@@ -0,0 +1,168 @@
+	config := mgr.GetConfig()
+
+	dcClient, _ := deploymentconfig.NewForConfig(config)
+
+	// Check if DC already exists
+	existingDC, _ := dcClient.DeploymentConfigs(ServicesNamespace).Get(ServiceName, metav1.GetOptions{})
+
+	if existingDC.Name == ServiceName {
+		return &ReconcileService{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+	}
+
+	// If DC didn't exist, then we can skip
+	_, err := dcClient.DeploymentConfigs(ServicesNamespace).Create(newDeploymentConfigForAppService(nil, ServiceName, ServicesNamespace))
+	if err != nil {
+		fmt.Println(err) // Log.Error(..)
+	}
+
+	// TODO: Create a service, if absent
+
+	return &ReconcileService{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+
+	// Create a new controller
+	c, err := controller.New("topology-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Do not send events which are related to the specific DC.
+	pred := predicate.Funcs{
+		// TODO: When the deployment is being created, DC gets updated
+		// and that would trigget this. How do we filter out such events?
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaOld.GetName() == ServiceName && e.MetaOld.GetNamespace() == ServicesNamespace
+		},
+
+		// TODO: In all probability, any delele event is interesting to us.
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == ServiceName && e.Meta.GetNamespace() == ServicesNamespace
+		},
+
+		// TODO: When a new one is created because an operator is being deployed..
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: &v1.DeploymentConfig{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Reconcile handles events related to changes to the App Topology Service deployment.
+// This includes events from service/dc named "ServiceName" in the namespace "ServiceNameSpace"
+func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// TODO: Watch changes to the DeploymentConfig ( .. and the service, and the route )
+	// something happened to the specific DC, please react..
+	return reconcile.Result{}, nil
+}
+
+func newDeploymentConfigForAppService(containerPorts []corev1.ContainerPort, serviceName string, serviceNameSpace string) *v1.DeploymentConfig {
+	labels := getLabelsForServiceDeployments(ServiceName)
+	//annotations := resource.GetAnnotationsForCR(cp)
+	if containerPorts == nil {
+		containerPorts = []corev1.ContainerPort{{
+			ContainerPort: 8080,
+			Protocol:      corev1.ProtocolTCP,
+		}}
+	}
+	return &v1.DeploymentConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: ServicesNamespace,
+			Labels:    labels,
+			//	Annotations: annotations,
+		},
+		Spec: v1.DeploymentConfigSpec{
```

Comment:
Why not Deployment?

Reply:
Right. We could probably make it a `Deployment` since this is not going to use `ImageStream`s